p2p/types/request_map_test.go

```diff
@@ -0,0 +1,94 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRequestMap_Get(t *testing.T) {
+	rmap := NewRequestMap(100)
+
+	request := rmap.Get()
+
+	assert.Equal(
+		t,
+		request.Nonce(),
+		uint32(1),
+		"Request map error: failed to retrieve new request with valid nonce",
```

Comment:
I'm confused by how this check for request retrieval when we're comparing nonce values.

Reply:
So the nonce is monotonically increasing, correct?

Is this documented somewhere?